### PR TITLE
Search bar implementation

### PR DESCRIPTION
**Pull request by FranciscoGaraventa**
Changes made:
- Added a class that manage all the possible states when movies fetchs.
- The stream controller of movies_bloc now reveive elements of type
  search_event.
- Added a CustomScrollView with a SliverAppBar in movie_info_screen.

I add two gifs that show how search works when no results are found and when they are.

noResultFound

searchResult

This gif show how CustomScrollView works in the screen of movie information
scrollMovieInfo

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.0
+  flutter_bloc: ^6.1.1
```
Comment (MarcosCGlobant) [inline pubspec.yaml]: Why did you add this dependency? Where do you need to use that package?

Comment (FranciscoGaraventa) [reply pubspec.yaml]: I don't need it. I tried to use it in a solution that I had in mind but in the end I went the other way. I removed the dependency.

**Review by MarcosCGlobant — APPROVED** [review]
LGTM
